### PR TITLE
add a 'size' parameter to Cursor.stream()

### PR DESCRIPTION
**Pull request by dlax**
Based on #793

### CODE REVIEW

File: docs/api/cursors.rst [hunk 1]
```diff
@@ -180,7 +180,14 @@ The `!Cursor` class
         .. __: https://materialize.com/docs/sql/subscribe/
         .. __: https://www.cockroachlabs.com/docs/stable/changefeed-for.html
 
-        The parameters are the same of `execute()`.
+        The parameters are the same of `execute()`, except for `size` which
+        can be used to set results retrieval by chunks instead of row-by-row.
+
+        .. note::
+
+            This `size` parameter is only available from libpq 17, you can use
```
Comment (dvarrazzo) [inline docs/api/cursors.rst:188]:
```suggestion
            This `size` parameter is only available from libpq 17, you can use the
```

**Comment by dvarrazzo** [comment]
Heads up: I gave a forced push to the branch because github didn't detect that #793 was merged and the MR still showed two commits.

File: psycopg/psycopg/cursor_async.py [hunk 2]
```diff
@@ -138,23 +138,32 @@ async def stream(
...
-                    self._stream_send_gen(query, params, binary=binary)
+                    self._stream_send_gen(query, params, binary=binary, size=size)
                 )
                 first = True
                 while await self._conn.wait(self._stream_fetchone_gen(first)):
-                    # We know that, if we got a result, it has a single row.
-                    rec: Row = self._tx.load_row(0, self._make_row)  # type: ignore
-                    yield rec
+                    # We know that, if we got a result, it has a single row or
+                    # a chunk of rows.
```
Comment (dvarrazzo) [inline psycopg/psycopg/cursor_async.py:161]: I think we can drop this comment altogether. The previous one was about assumption about there being exactly one record, but here we are making a careful iteration on >= 0 records.
```suggestion
```

File: docs/api/cursors.rst (hunk 1 above)
Comment (dvarrazzo) [inline docs/api/cursors.rst:188]:
```suggestion
            This `size` parameter is only available from libpq 17, you can use the
```

**Review by dvarrazzo — CHANGES_REQUESTED** [review]
Just some nits, but otherwise looks good, thank you :slightly_smiling_face:

File: psycopg/psycopg/_cursor_base.py [hunk 3]
```diff
@@ -329,7 +338,7 @@ def _stream_fetchone_gen(self, first: bool) -> PQGen[Optional["PGresult"]]:
             return None
 
         status = res.status
-        if status == SINGLE_TUPLE:
+        if status in (SINGLE_TUPLE, TUPLES_CHUNK):
```
[the remark this answers is not in the document]
Comment (dlax) [reply psycopg/psycopg/_cursor_base.py]: I didn't know that; thank you for sharing.

Fixed, along with other suggestions.

**Comment by dvarrazzo** [comment]
I've rebased this to master and about to merge it.

**Pull request merged**